Signal/src/ViewControllers/MediaPageViewController.swift

```diff
@@ -0,0 +1,706 @@
+
+        let kSpacingBetweenItems: CGFloat = 20
+
+        super.init(transitionStyle: .scroll,
+                   navigationOrientation: .horizontal,
+                   options: [UIPageViewControllerOptionInterPageSpacingKey: kSpacingBetweenItems])
+
+        self.dataSource = self
+        self.delegate = self
+
+        uiDatabaseConnection.beginLongLivedReadTransaction()
+
+        if includeGallery {
+            uiDatabaseConnection.read { transaction in
+                // TODO don't read all media messages in at once. Use Mapping?
+                self.mediaGalleryFinder.enumerateMediaMessages(with: thread, transaction: transaction) { message in
+                    self.mediaMessages.append(message)
+                }
+            }
+        } else {
+            self.mediaMessages = [mediaMessage]
+        }
+
+        guard let initialItem = self.buildGalleryItem(mediaMessage: mediaMessage, thread: thread) else {
+            owsFail("unexpetedly unable to build initial gallery item")
+            return
+        }
+        self.initialItem = initialItem
+        cachedItems.insert(initialItem, at: 0)
+
+        self.setViewControllers([initialItem.viewController], direction: .forward, animated: false, completion: nil)
+    }
+
+    @available(*, unavailable, message: "Unimplemented")
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    var presentationView: UIImageView!
+    var footerBar: UIToolbar!
+    var videoPlayBarButton: UIBarButtonItem!
+    var videoPauseBarButton: UIBarButtonItem!
+    var pagerScrollView: UIScrollView!
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+
+        // Navigation
+
+        self.navigationItem.leftBarButtonItem = UIBarButtonItem(barButtonSystemItem: .stop, target: self, action: #selector(didPressDismissButton))
+
+        // Even though bars are opaque, we want content to be layed out behind them.
+        // The bars might obscure part of the content, but they can easily be hidden by tapping
+        // The alternative would be that content would shift when the navbars hide.
+        self.extendedLayoutIncludesOpaqueBars = true
+        self.automaticallyAdjustsScrollViewInsets = false
+
+        // Get reference to paged content which lives in a scrollView created by the superclass
+        // We show/hide this content during presentation
+        for view in self.view.subviews {
+            if let pagerScrollView = view as? UIScrollView {
+                self.pagerScrollView = pagerScrollView
+            }
+        }
+
+        // Hack to avoid "page" bouncing when not in gallery view.
+        // e.g. when getting to media details via message details screen, there's only
+        // one "Page" so the bounce doesn't make sense.
+        if !self.includeGallery {
+            pagerScrollView.isScrollEnabled = false
+        }
+
+        // FIXME dynamic title with sender/date
+        self.title = "Attachment"
```

Comment:
l10n?

Reply:
Or is this definitely changing in next PR?

Reply:
I'm going to get this in a followup PR.